test/e2e/set/scyllacluster/scyllacluster_tls.go

```diff
@@ -260,6 +265,130 @@ var _ = g.Describe("ScyllaCluster", func() {
+		o.Expect(err).NotTo(o.HaveOccurred())
+		_, adminClientCertBytes, _, adminClientKeyBytes := verification.VerifyAndParseTLSCert(adminClientSecret, verification.TLSCertOptions{
+			IsCA:     pointer.Ptr(false),
+			KeyUsage: pointer.Ptr(x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature),
+		})
+
+		servingCAPool := x509.NewCertPool()
+		servingCAPool.AddCert(servingCACerts[0])
+		adminTLSCert, err := tls.X509KeyPair(adminClientCertBytes, adminClientKeyBytes)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		const loaders = 3
+
+		for i := 0; i < loaders; i++ {
+			clusterConfig := gocql.NewCluster(hosts...)
+			clusterConfig.Timeout = 3 * time.Second
+			clusterConfig.ConnectTimeout = 3 * time.Second
+			clusterConfig.Port = 9142
+			clusterConfig.SslOpts = &gocql.SslOptions{
+				Config: &tls.Config{
+					ServerName:         host,
+					InsecureSkipVerify: false,
+					Certificates:       []tls.Certificate{adminTLSCert},
+					RootCAs:            servingCAPool,
+				},
+			}
+
+			di := insertAndVerifyCQLData(ctx, hosts, utils.WithClusterConfig(clusterConfig))
+			defer di.Close()
+
+			trafficCtx, trafficCtxCancel := context.WithCancel(ctx)
+			defer trafficCtxCancel()
+
+			stopTraffic, err := di.StartContinuousReads(trafficCtx)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			defer func() {
+				err := stopTraffic()
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}()
+		}
+
+		framework.By("Adding new DNS domain to rotate certificates")
+		testDNSDomain := fmt.Sprintf("%s.scylla-operator.scylladb.com", rand.String(6))
+
+		sc.ManagedFields = nil
+		sc.ResourceVersion = ""
+		sc.Spec.DNSDomains = append(sc.Spec.DNSDomains, testDNSDomain)
+		scData, err := runtime.Encode(scheme.Codecs.LegacyCodec(scyllav1.GroupVersion), sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// TODO: Use generated Apply method when our clients have it.
+		//       Ref: https://github.com/scylladb/scylla-operator/issues/1474
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(ctx, sc.Name, types.ApplyPatchType, scData, metav1.PatchOptions{
+			FieldManager: f.FieldManager(),
+			Force:        pointer.Ptr(true),
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Starting to probe node %q for updated certs", host)
+
+		o.Eventually(func(eo o.Gomega) {
+			serverCerts, err := utils.GetServerTLSCertificates(fmt.Sprintf("%s:9142", host), &tls.Config{
+				ServerName:         host,
+				InsecureSkipVerify: false,
+				Certificates:       []tls.Certificate{adminTLSCert},
+				RootCAs:            servingCAPool,
+			})
+			eo.Expect(err).NotTo(o.HaveOccurred())
+			eo.Expect(serverCerts).NotTo(o.BeEmpty())
+
+			eo.Expect(serverCerts[0].DNSNames).To(o.Satisfy(func(dnsNames []string) bool {
+				return slices.Contains(dnsNames, func(dnsName string) bool {
+					return strings.HasSuffix(dnsName, testDNSDomain)
```

Comment:
why test for HasSuffix instead of equality?

Reply:
because cert is also signed for individual nodes. We need to detect cert change, so when test domain is added, it means it changed.